content/en/docs/concepts/scheduling-eviction/taint-and-toleration.md

```diff
@@ -85,8 +85,7 @@ A toleration "matches" a taint if the keys are the same and the effects are the
 
 There are two special cases:
 
-An empty `key` with operator `Exists` matches all keys, values and effects which means this
-will tolerate everything.
+If the `key` is empty, then the `operator` must be `Exists`, which matches all keys and values. Note that the `effect` still needs to be matched at the same time.
```

Comment:
are you sure `effect` is used for matching?

Reply:
Yes, according to the [implementation](https://github.com/kubernetes/kubernetes/blob/af2bf2d8f1b3437fe67e1686266a176320422465/staging/src/k8s.io/api/core/v1/toleration.go#L38). It will always check whether if effect is matched at first.
```go

// ToleratesTaint checks if the toleration tolerates the taint.
// The matching follows the rules below:
//
//  1. Empty toleration.effect means to match all taint effects,
//     otherwise taint effect must equal to toleration.effect.
//  2. If toleration.operator is 'Exists', it means to match all taint values.
//  3. Empty toleration.key means to match all taint keys.
//     If toleration.key is empty, toleration.operator must be 'Exists';
//     this combination means to match all taint values and all taint keys.
func (t *Toleration) ToleratesTaint(taint *Taint) bool {
	if len(t.Effect) > 0 && t.Effect != taint.Effect {
		return false
	}
        ...
}
```